### PR TITLE
move ThreadExecuteTimings to solana-ledger

### PR DESCRIPTION
**Pull request by kevinheavey**
See item 1 in #480:

> Move ThreadExecuteTimings to solana-ledger as it's not used anywhere else

### CODE REVIEW

**Review by buffalojoec — COMMENTED** [review]
Lgtm, I think considering there's other timings defined in `blockstore_processor` it makes sense.

That said, even though this file isn't exactly super organized, do you mind sliding `ThreadExecuteTimings` down to live alongside the other timing definitions? I'm thinking right below `BatchExecutionTiming`.

Also, CI has a lockfile check so, as the message says, just run `./scripts/cargo-for-all-lock-files.sh tree` and commit the result.

File: ledger/src/blockstore_processor.rs [hunk 1]
```diff
@@ -214,6 +218,38 @@ pub fn execute_batch(
...
+    pub total_thread_us: u64,
+    pub total_transactions_executed: u64,
+    pub execute_timings: ExecuteTimings,
+}
+
+impl ThreadExecuteTimings {
+    pub fn report_stats(&self, slot: Slot) {
+        lazy! {
+            datapoint_info!(
+                "replay-slot-end-to-end-stats",
```
Comment (steviez) [inline ledger/src/blockstore_processor.rs]: I don't want to punish a good deed 😉 , but it has annoyed me that these datapoints (this one and the one below) both mention `replay` specifically, but neither live with replay code:
https://github.com/anza-xyz/agave/blob/70c4cb0ba1fbef2b3aa5796420ed285c46f5c4f6/core/src/consensus/progress_map.rs#L49-L50

Granted the one I just linked is `ReplaySlotStats` which is technically `ReplayStage` specific, but `ReplaySlotStats` is an aliased `ConfirmationTiming` (which is in `blockstore_processor`) with a few extra values that get passed in from `ReplayStage` main loop.

In any case, the one I mentioned invokes the one you're moving here:
https://github.com/anza-xyz/agave/blob/70c4cb0ba1fbef2b3aa5796420ed285c46f5c4f6/core/src/consensus/progress_map.rs#L86

I think it'd be nice to have these next to each other in code, potentially both in `blockstore_processor`. Would you be up for moving this other one I mentioned as well ?

**Comment by steviez** [comment]
Also, you had a CI failure. Since you updated `/Cargo.lock`, you should run `scripts/cargo-for-all-lock-files.sh tree` which will update `/programs/sbf/Cargo.lock`; I never remember to run this script either ha 😅

**Comment by kevinheavey** [comment]
^ All done

**Review by buffalojoec — DISMISSED** [review]
Nice, thanks!

File: ledger/src/lib.rs [hunk 2]
```diff
@@ -1,5 +1,6 @@
 #![cfg_attr(RUSTC_WITH_SPECIALIZATION, feature(min_specialization))]
 #![allow(clippy::arithmetic_side_effects)]
+#![recursion_limit = "2048"]
```
Comment (steviez) [inline ledger/src/lib.rs:3]: This was just a precaution right; we didn't actually bump up against this / this would cause compile time break right ?

Comment (kevinheavey) [reply ledger/src/lib.rs:3]: It's real, without it `cargo +nightly-2024-01-05` throws this error:
```
error: recursion limit reached while expanding `eager_internal!`
    --> ledger/src/blockstore_processor.rs:1153:9
     |
1153 | /         lazy! {
1154 | |             datapoint_info!(
1155 | |                 "replay-slot-end-to-end-stats",
1156 | |                 ("slot", slot as i64, i64),
...    |
1162 | |             );
1163 | |         };
```

Comment (kevinheavey) [reply ledger/src/lib.rs:3]: On my machine `#![recursion_limit = "1247"]` is the lowest limit that makes it compile, if that's any use to you

Comment (kevinheavey) [reply ledger/src/lib.rs:3]: Also now that this code is out of program-runtime, program-runtime now compiles if I remove the `#![recursion_limit = "2048"]` in its lib.rs so I'm removing it from there

Comment (steviez) [reply ledger/src/lib.rs:3]: https://github.com/Emoun/eager/blob/58b04fc1f06d624cc81825b1a4d1b41330d6e412/src/eager.rs#L45-L48
😬

**Review by steviez — APPROVED** [review]
Will let CI run, but this LGTM. Thanks!

**Pull request merged**